test/fuzz/utility.h

```diff
@@ -52,6 +52,19 @@ inline Protobuf::RepeatedPtrField<envoy::api::v2::core::HeaderValueOption> repla
   return processed;
 }
 
+inline envoy::api::v2::core::Metadata
+replaceInvalidStringValues(const envoy::api::v2::core::Metadata& upstream_metadata) {
+  envoy::api::v2::core::Metadata processed = upstream_metadata;
+  for (auto& metadata_struct : *processed.mutable_filter_metadata()) {
+    for (auto& field : *metadata_struct.second.mutable_fields()) {
+      if (field.second.kind_case() == ProtobufWkt::Value::kStringValue) {
```

Comment:
Do you think we need to handle other recursive types (struct, list, etc.) here?

https://github.com/protocolbuffers/protobuf/blob/2d61b9edd95a688a720132cde6af536fd5b5dc7a/src/google/protobuf/struct.proto#L73-L76

Reply:
I wasn't too concerned with this as an issue to fuzz over since metadata usually comes from the config (e.g. listenerMetadata(), or routeEntry metadata) or from the connection's streaminfo, so it'd be on the onus of the one configuring Envoy to make sure there's no NUL characters. I would have handled them only if the fuzzer gets blocked on them, but I'm not sure how complicated the fuzzer will get for this case. I could handle them, though, if you think it'd be easier to get done in one PR

Reply:
I think we can skip this for now then. @junr03 thoughts?

Reply:
hey! just a ping, would love to move forward. 
thanks!

Reply:
@asraa could you please leave a note related to the discussion above? Thanks!

Reply:
Done! Thank you.